src/wasm-emscripten.h

```diff
@@ -18,27 +18,47 @@
 #define wasm_wasm_emscripten_h
 
 #include "wasm.h"
+#include "wasm-builder.h"
 
 namespace wasm {
 
 class LinkerObject;
 
-namespace emscripten {
+class EmscriptenGlueLinker {
+public:
+  EmscriptenGlueLinker(Module& _wasm, Address _stackPointerOffset)
+    : wasm(_wasm),
+      builder(_wasm),
+      stackPointerOffset(_stackPointerOffset) { }
 
-void generateRuntimeFunctions(LinkerObject& linker);
-void generateMemoryGrowthFunction(Module&);
+  void generateRuntimeFunctions();
+  void generateMemoryGrowthFunction();
 
-// Create thunks for use with emscripten Runtime.dynCall. Creates one for each
-// signature in the indirect function table.
-std::vector<Function*> makeDynCallThunks(Module& wasm, std::vector<Name> const& tableSegmentData);
+  // Create thunks for use with emscripten Runtime.dynCall. Creates one for each
+  // signature in the indirect function table.
```

Comment:
Does it add them directly to the module? If so, why return them also?

Reply:
Edit: OK I see it's to export them. Maybe `makeDynCallThunks` could just export them itself, right after creating them?

Reply:
That's probably best yeah.